test/reorder/rcm.cpp

```diff
@@ -0,0 +1,395 @@
+                    if (levels[x_neighbour] == level - 1) {
+                        if (inv_perm[x_neighbour] <
+                            inv_perm[x_first_neighbour]) {
+                            x_first_neighbour = x_neighbour;
+                        }
+                    }
+                }
+                // Same again, for y.
+                auto y_first_neighbour = perm[n - 1];
+                const auto y_row_start = row_ptrs[y];
+                const auto y_row_end = row_ptrs[y + 1];
+                for (auto y_neighbour_idx = y_row_start;
+                     y_neighbour_idx < y_row_end; ++y_neighbour_idx) {
+                    const auto y_neighbour = col_idxs[y_neighbour_idx];
+                    if (levels[y_neighbour] == level - 1) {
+                        if (inv_perm[y_neighbour] <
+                            inv_perm[y_first_neighbour]) {
+                            y_first_neighbour = y_neighbour;
+                        }
+                    }
+                }
+
+                // Assert the ... is not after the ... in the previous level.
+                ASSERT_GE(inv_perm[y_first_neighbour],
+                          inv_perm[x_first_neighbour])
+                    << "First neighbor ordering violated between nodes " << x
+                    << " and " << y << ", first neighbors were "
+                    << x_first_neighbour << " and " << y_first_neighbour;
+
+                if (y_first_neighbour == x_first_neighbour) {
+                    ASSERT_GE(degrees[y], degrees[x])
+                        << "Degree ordering violated between nodes " << x
+                        << " and " << y << ", degrees were " << degrees[x]
+                        << " and " << degrees[y];
+                }
+            }
+        }
+    }
+
+    void build_multiple_connected_components()
+    {
+        gko::matrix_data<value_type, index_type> data;
+        d_1138_bus_mtx->write(data);
+        const auto num_rows = data.size[0];
+        const auto nnz = data.nonzeros.size();
+        const int num_copies = 5;
+        data.size[0] *= num_copies;
+        data.size[1] *= num_copies;
+        for (gko::size_type i = 0; i < nnz; i++) {
+            const auto entry = data.nonzeros[i];
+            // create copies of the matrix
+            for (int copy = 1; copy < num_copies; copy++) {
+                data.nonzeros.emplace_back(entry.row + copy * num_rows,
+                                           entry.column + copy * num_rows,
+                                           entry.value);
+            }
+        }
+        std::vector<index_type> permutation(data.size[0]);
+        std::iota(permutation.begin(), permutation.end(), 0);
+        // std::shuffle(permutation.begin(), permutation.end(), rng);
+        for (auto& entry : data.nonzeros) {
+            entry.row = permutation[entry.row];
+            entry.column = permutation[entry.column];
+        }
```

Comment:
it is not shuffled

Reply:
oh good catch, I disabled it when debugging and forgot to reenable it